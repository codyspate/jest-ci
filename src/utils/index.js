export const someOtherDevChangesTheLoan = loan => {
  loan.set({
    a: 1,
    b: 1
  });
};

export const setToValue = (obj, value, path) => {
  let i;
  path = path.split('.');
  for (i = 0; i < path.length - 1; i += 1) obj = obj[path[i]];
  obj[path[i]] = value;
};

export const getValue = (obj, path) => {
  let i;
  path = path.split('.');
  for (i = 0; i < path.length - 1; i += 1) obj = obj[path[i]];
  return obj[path[i]];
};
