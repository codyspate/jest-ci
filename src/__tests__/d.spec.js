import { addAandB } from '../index';
import { someOtherDevChangesTheLoan } from '../utils';
import Loan from '../__mocks__/loan';

// Pretend its a DB

const loan = new Loan();

loan.set({
  a: 2,
  b: 3
});

describe('Deeeeeee', () => {
  // it('Not doing anything', () => {});
  it('Should add two things', () => {
    someOtherDevChangesTheLoan(loan);
    expect(addAandB(loan)).toBe(5);
  });
});
