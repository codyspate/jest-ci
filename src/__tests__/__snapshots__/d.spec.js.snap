// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Examples validates snapshots 1`] = `
Object {
  "foo": "bar",
}
`;
